src/eos-updater-util.c

```diff
@@ -154,14 +211,147 @@ eos_updater_resolve_upgrade (EosUpdater  *updater,
 
   if (booted_checksum)
     *booted_checksum = g_strdup (booted);
 
+  /* Get branch configuration data baseurl */
+  repo_config = ostree_repo_get_config (repo);
+  osgroup = g_strdup_printf ("remote \"%s\"", osname);
+  baseurl = g_key_file_get_string (repo_config, osgroup, "url", NULL);
+  if (!baseurl)
+    {
+      g_set_error (error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND,
+                   "No url found for %s, cannot upgrade", osname);
+      goto out;
+    }
+
+  /* Get product identifier */
+  hw_descriptors = g_hash_table_new_full (g_str_hash, g_str_equal,
+                                          g_free, g_free);
+  if (g_file_test (DT_COMPATIBLE, G_FILE_TEST_EXISTS))
+    { /* ARM */
+      gs_unref_object GFile *fp = NULL;
+      gs_free gchar *fc = NULL;
+
+      fp = g_file_new_for_path (DT_COMPATIBLE);
+      if (g_file_load_contents (fp, NULL, &fc, NULL, NULL, NULL))
+        {
+          gs_strfreev gchar **sv = g_strsplit (fc, ",", -1);
+
+          if (sv && sv[0])
+            g_hash_table_insert (hw_descriptors, g_strdup ("sys_vendor"),
+                                 g_strdup (g_strstrip (sv[0])));
+          if (sv && sv[1])
+            g_hash_table_insert (hw_descriptors, g_strdup ("product_name"),
+                                 g_strdup (g_strstrip (sv[1])));
+        }
+    }
+  else
+    { /* X86 */
+      guint i;
+
+      for (i = 0; dmi_attributes[i]; i++)
+        {
+          gs_free gchar *path = NULL;
+          gs_unref_object GFile *fp = NULL;
+          gs_free gchar *fc = NULL;
+          gsize len;
+
+          path = g_strconcat (DMI_PATH, dmi_attributes[i], NULL);
+          fp = g_file_new_for_path (path);
+          if (g_file_load_contents (fp, NULL, &fc, &len, NULL, NULL))
+          {
+              if (len > 128)
+                fc[128] = '\0';
+              g_hash_table_insert (hw_descriptors, g_strdup (dmi_attributes[i]),
+                                   g_strdup (g_strstrip (fc)));
+          }
+        }
+    }
+
+  if (!g_hash_table_lookup (hw_descriptors, "sys_vendor"))
+    g_hash_table_insert (hw_descriptors, g_strdup ("sys_vendor"),
+                         g_strdup ("EOSUNKNOWN"));
+
+  if (!g_hash_table_lookup (hw_descriptors, "product_name"))
+    g_hash_table_insert (hw_descriptors, g_strdup ("product_name"),
+                         g_strdup ("EOSUNKNOWN"));
+
+  vendor = g_strdup (g_hash_table_lookup (hw_descriptors, "sys_vendor"));
+  product = g_strdup (g_hash_table_lookup (hw_descriptors, "product_name"));
+  p_id = g_strconcat (cleanstr (vendor), " ", cleanstr (product), NULL);
+  message ("Product group: %s", p_id);
+
+  /* Build the HTTP URI */
+  g_hash_table_insert (hw_descriptors, g_strdup ("ref"), g_strdup (o_ref));
+  g_hash_table_insert (hw_descriptors, g_strdup ("commit"), g_strdup (booted));
+  query = soup_form_encode_hash (hw_descriptors);
+  uri = g_strconcat (baseurl, "/", BRANCHES_CONFIG_PATH, "?", query, NULL);
```

Comment:
I could, but I didn't see any clear advantage of one over the other when checking libsoup's API. Plus, I was put off of diving more into the differences by this comment in SoupURI's documentation:

> Many applications will not need to use SoupURI directly at all; on the client side, soup_message_new() takes a stringified URI (...)

Please alert me if we would be gaining something relevant by switching to SoupURI, as I'm not an experienced libsoup user, but if it is just a matter of preference I would rather keep it this way.